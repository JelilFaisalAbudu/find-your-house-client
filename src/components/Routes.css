.site-header {
  position: absolute;
  padding: 1.5em 0;
  width: 100%;
  z-index: 1020;
}

.navbar-nav .nav-item:not(:last-child) {
  margin-right: 2em;
}

.navbar-nav .nav-link {
  color: #fff !important;
  border-radius: 2.4em;
  margin-top: 1em;
  margin-bottom: 1em;
  padding: 0.5em 1.5em !important;
  text-transform: uppercase;
}

.nav-link.active,
.nav-link:hover {
  background-color: #7b7b77;
}

.navbar-brand .brand-logo {
  height: 4rem;
}

.navbar-brand:hover {
  background-color: #f037099b;
}

.remove-absolute {
  background-color: currentColor;
  position: sticky;
  top: 0;
  z-index: 100000;
  transition: position 0.3s;
}

.footer__social {
  background-color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 1em 0;
}

.footer-icons {
  margin: 0.7em;
}

.fa,
.fab {
  font-size: 2rem;
}

.fa-twitter {
  background-color: #00aff1;
  color: #fff;
}

.fa-github {
  background-color: rgb(83, 82, 82);
  color: #fff;
}

.fa-linkedin-in {
  color: white;
  background-color: #0a66c2;
}

.footer-icons i {
  font-size: 1.8rem;
  border-radius: 50%;
  padding: 0.3em;
}

.footer-icons i:hover {
  background-color: #7b7b77;
}

.copyright__copy {
  vertical-align: middle;
}

@media (min-width: 768px) {
  .developed-by {
    display: block;
  }
}

@media (max-width: 991px) {
  .site-header--bg {
    background-color: currentColor;
    opacity: 0.9;
  }

  .site-header {
    position: unset;
  }

  .navbar-toggler {
    color: white;
    background-color: white;
  }
}
