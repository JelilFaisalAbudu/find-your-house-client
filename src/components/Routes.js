/* eslint-disable no-unused-vars */
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  Router,
  Switch,
  Route,
  Link,
} from 'react-router-dom';

import history from '../helpers/history';
import { logout } from '../redux/actions/auth';
import message from '../redux/actions/message';

import Login from './Login';
import Register from './Register';
import Home from './Home';
import Favorites from './Favorites';
import BoardUser from './BoardUser';

const Routes = () => {
  const { user: currentUser } = useSelector(state => state.auth);
  const dispatch = useDispatch();

  useEffect(() => {
    history.listen(() => {
      dispatch(message.clearMessage()); // clear message when changing location
    });
  }, [dispatch]);

  const logOut = () => {
    dispatch(logout());
  };

  return (
    <Router history={history}>
      <div>
        <nav className="navbar navbar-expand navbar-dark bg-dark">
          <Link to="/" className="navbar-brand">
            NavBar Brand
          </Link>
          <div className="navbar-nav mr-auto">
            <li className="nav-item">
              <Link to="/home" className="nav-link">
                Home
              </Link>
            </li>
          </div>

          {currentUser ? (
            <div className="navbar-nav ml-auto">
              <li className="nav-item">
                <Link to="/Favorites" className="nav-link">
                  {currentUser.name}
                </Link>
              </li>
              <li className="nav-item">
                <Link to="/logout" className="nav-link" onClick={logOut}>
                  LogOut
                </Link>
              </li>
            </div>
          ) : (
            <div className="navbar-nav ml-auto">
              <li className="nav-item">
                <Link to="/login" className="nav-link">
                  Login
                </Link>
              </li>

              <li className="nav-item">
                <Link to="/register" className="nav-link">
                  Sign Up
                </Link>
              </li>
            </div>
          )}
        </nav>

        <div className="container mt-3">
          <Switch>
            <Route exact path={['/', '/home', '/logout']} component={Home} />
            <Route exact path="/login" component={Login} />
            <Route exact path="/register" component={Register} />
            <Route exact path="/Favorites" component={Favorites} />
            <Route path="/user" component={BoardUser} />
          </Switch>
        </div>
      </div>
    </Router>
  );
};

export default Routes;
