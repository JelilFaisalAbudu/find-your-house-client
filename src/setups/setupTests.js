import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-17-updated';

Enzyme.configure({ adapter: new Adapter() });
