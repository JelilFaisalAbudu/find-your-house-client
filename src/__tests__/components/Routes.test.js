import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import '../setups/setupTests';
import Routes from '../../components/Routes';
import store from '../../redux/store';

const setup = () => mount(
  <Provider store={store}>
    <Routes />
  </Provider>,
);

describe('Routes/Navbar', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = setup();
  });

  test('renders the Sign-up page without crashing', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  describe('Unauthenticated User', () => {
    let navLinks;

    beforeAll(() => {
      navLinks = wrapper.find('Link');
    });

    test('does not display the logout link with text Logout', () => {
      expect(navLinks.find({ to: '/logout' }).length).toEqual(0);
    });

    test('renders the login-in link with text Login', () => {
      expect(navLinks.find({ to: '/login' }).text()).toMatch(/Login/i);
    });

    test('renders the sign-up link with text Signup', () => {
      expect(navLinks.find({ to: '/register' }).text()).toMatch(/Sign Up/i);
    });
  });

  describe('Authenticated User', () => {
    let navLinks;
    const user = {
      id: 4,
      name: 'John Doe',
      auth_token: 'doiifhfieiffhghwnc4429hBD',
    };

    const registerAction = user => ({
      type: 'REGISTER_SUCCESS',
      payload: { user },
    });

    beforeAll(() => {
      store.dispatch(registerAction(user));
      wrapper = setup();
      navLinks = wrapper.find('Link');
    });

    test('does not display the login link with text Login', () => {
      expect(navLinks.find({ to: '/login' }).length).toEqual(0);
    });

    test(' does not render the sign-up link with text Signup', () => {
      expect(navLinks.find({ to: '/register' }).length).toEqual(0);
    });

    test('renders the link to the user"s favorites ', () => {
      expect(navLinks.find({ to: '/Favorites' }).length).toEqual(1);
    });

    test('renders the nav link for the logged-in user to log out', () => {
      expect(navLinks.find({ to: '/logout' }).text()).toMatch(/Logout/i);
    });
  });
});
