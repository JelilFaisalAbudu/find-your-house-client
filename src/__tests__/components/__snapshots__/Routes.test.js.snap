// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes/Navbar renders the Sign-up page without crashing 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Routes>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div>
        <nav
          className="navbar navbar-expand navbar-dark bg-dark"
        >
          <Link
            className="navbar-brand"
            to="/"
          >
            <LinkAnchor
              className="navbar-brand"
              href="/"
              navigate={[Function]}
            >
              <a
                className="navbar-brand"
                href="/"
                onClick={[Function]}
              >
                NavBar Brand
              </a>
            </LinkAnchor>
          </Link>
          <div
            className="navbar-nav mr-auto"
          >
            <li
              className="nav-item"
            >
              <Link
                className="nav-link"
                to="/home"
              >
                <LinkAnchor
                  className="nav-link"
                  href="/home"
                  navigate={[Function]}
                >
                  <a
                    className="nav-link"
                    href="/home"
                    onClick={[Function]}
                  >
                    Home
                  </a>
                </LinkAnchor>
              </Link>
            </li>
          </div>
          <div
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item"
            >
              <Link
                className="nav-link"
                to="/login"
              >
                <LinkAnchor
                  className="nav-link"
                  href="/login"
                  navigate={[Function]}
                >
                  <a
                    className="nav-link"
                    href="/login"
                    onClick={[Function]}
                  >
                    Login
                  </a>
                </LinkAnchor>
              </Link>
            </li>
            <li
              className="nav-item"
            >
              <Link
                className="nav-link"
                to="/register"
              >
                <LinkAnchor
                  className="nav-link"
                  href="/register"
                  navigate={[Function]}
                >
                  <a
                    className="nav-link"
                    href="/register"
                    onClick={[Function]}
                  >
                    Sign Up
                  </a>
                </LinkAnchor>
              </Link>
            </li>
          </div>
        </nav>
        <div
          className="container mt-3"
        >
          <Switch>
            <Route
              component={[Function]}
              computedMatch={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              exact={true}
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              path={
                Array [
                  "/",
                  "/home",
                  "/logout",
                ]
              }
            >
              <Home
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "createHref": [Function],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <div
                  className="houses-container"
                >
                  <div>
                    Loading...
                  </div>
                </div>
              </Home>
            </Route>
          </Switch>
        </div>
      </div>
    </Router>
  </Routes>
</Provider>
`;
